### ISSUE TITLE
Support Custom Content Loaders to Tap Into Builder and Bundler Plugin Ecosystems

### ISSUE THREAD

**Issue by peter-wasilko** [imba/imba#820]
I have been trying to a find a way to directly import YAML files into my IMBA projects. 

I have tried using **import {default as ViteYaml} from '@modyfi/vite-plugin-yaml';** in a vite.config.js file which lets Vite deal with YAML, but then the esbuild powered imba subcommands choke on the .yaml assets.

Conversely, the Vite dev server chokes on some imba esbuild/rollup supported asset types throwing a **SyntaxError: Unexpected token '{'** error in Safari dev tools console. 

So I am currently using Vite to bundle and Imba -w to test which is sub-optimal.

Since we have Vite, esBuild, and Rollup all at play we probably need a high level config file to let us map asset types to builder/bundler plugin names and configuration settings targeted to each tool unless imba could detect said plugins from the contents of package.json and auto-active them.

**Comment by familyfriendlymikey** [comment]
I agree that the current situation isn't ideal. Though I am wondering, if you're developing and building your project with Vite, what is the use for the esbuild powered imba subcommands? Which asset types does the Vite dev server choke on? And what is the nature of the testing you mentioned?

**Comment by peter-wasilko** [comment]
By test, I mean to use the HMR in the imba native developer server, since the Vite development server in the **imba 229** _imba create_ vite template generates boilerplate that throws an **Unexpected token '{'** on **Line 37** of **main.imba**. Then I started looking for other project templates I could tweak in the hope of arriving at a stable imba / vite / electron / electron-forge boilerplate that would support HMR development inside Electron since my project depends on some backend API's and Node modules that a browser based development preview wouldn't have access to. But I haven't found anything yet the can both automatically build an electron based desktop app and deal with imba's importation of asset types that vite doesn't directly support. 

For example, one of the dead end templates that I tried would cause vite to complain that it couldn't import css files without a loader, while the dev server inside imba wouldn't pick up on the vite yaml loading plugin. So vite bundled imba can directly load yaml, but imba served imba doesn't recognize yaml.

And if I try 

 ▶︎ imba --watch --vite ./index.html
  
I get:

✘ [ERROR] Cannot start service: Host version "0.16.17" does not match binary version "0.17.10"

1 error

Perhaps all the requisite functionally is already in place and it is just a matter of some advanced configuration file magic that I haven't mastered as yet.

**Comment by familyfriendlymikey** [comment]
Is the Vite project generated with imba@229 create only throwing when you use electron? More specifically, are there any issues with the following?:

```
npm i -g imba
imba create -t vite test-vite
cd test-vite
npm run dev
```

I could be wrong, but I think there should be a working configuration here with Vite and Electron that is able to import any type of asset provided there is a corresponding Vite plugin, without using Imba's bundler. Imba's bundler isn't aware of any Vite configuration, it's completely separate from our `vite-plugin-imba`.

It would be interesting to get a template set up. Have you looked into something like this? https://github.com/electron-vite/electron-vite-vue/blob/main/vite.config.ts I don't use Electron myself but I might be able to give it a shot.

**Comment by peter-wasilko** [comment]
Out of the box, the test project throws a **SyntaxError: Unexpected token '{',** but it turns out this only happens when the dev server is access from Safari on OS X, Version 16.3 (17614.4.6.11.6, 17614)

So we have a Safari specific bug in the boilerplate or a bona fides bug in Safari itself!

More interestingly, the test project does not throw the error in Safari Technology Preview: Release 164 (Safari 16.4, WebKit 17616.1.2)

**Comment by familyfriendlymikey** [comment]
Ah, right, Vite isn't working in dev mode with Safari right now https://github.com/imba/imba/issues/782

Seems like this is the root of the matter then? Assuming you have a Vite template set up with Electron, just for diagnosis, does `vite build --watch` meet all your requirements except for lack of actual HMR? It should still live reload but not hot replace.

**Comment by peter-wasilko** [comment]
> Seems like this is the root of the matter then?

I think so, I'll try to scaffold out a revised electron template and report back. Thanks so much for helping me sort through this!

**Comment by peter-wasilko** [comment]
I have made a further observation that some of the Electron related npm modules will fail rather cryptically if one is using pnpm instead of npm as one's package manager because they don't anticipate pnpm's use of file links to conserve storage space. 

Perhaps we could mention this on the imba documentation website to insure that anyone experiencing such issues doesn't mistakenly waste time exploring imba starter template code as a potential source of their woes.
